src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicHelpers.cs

```diff
@@ -56,29 +56,37 @@ internal static unsafe QuicAddr ToQuicAddr(this IPEndPoint iPEndPoint)
     }
 
     internal static unsafe T GetMsQuicParameter<T>(MsQuicSafeHandle handle, uint parameter)
-        where T: unmanaged
+        where T : unmanaged
     {
         T value;
         uint length = (uint)sizeof(T);
 
-        ThrowHelper.ThrowIfMsQuicError(MsQuicApi.Api.ApiTable->GetParam(
+        int status = MsQuicApi.Api.ApiTable->GetParam(
             handle.QuicHandle,
             parameter,
             &length,
-            (byte*)&value),
-            $"GetParam({handle}, {parameter}) failed");
+            (byte*)&value);
+
+        if (StatusFailed(status))
```

Comment:
It does not force creating the interpolated string (and avoids call to `SafeMsQuicHandle.ToString()` on success path)

Reply:
If we feel strongly about not creating `_traceId` we might add condition in `SafeMsQuicHandle.ToString`. If we don't, and we're ok allocating it in case of an error, we could add overload for `ThrowIfMsQuicError` that takes `FormattableString`.

I got the feeling that this will creep back in if we don't take precautions. What do you think?

Reply:
Can we use FormattableString in a way that does not cause excess allocations? I have never used it and the few tutorials I found look like the args are embedded in the instance (which would imply allocating one instance per call)

Reply:
FormattableString will incur its own costs: you won't get a string formatted, but you will get two allocations, for the FormattableString itself and for the object[] args array.

I think what you have here is fine.